### PR TITLE
Common - Add `setDead` API

### PR DESCRIPTION
**Pull request by johnb432**
**When merged this pull request will:**
- Title.
- Added to be used in #9988 and other.

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/common/functions/fnc_setDead.sqf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * Example:
+ * [cursorObject, player, player] call ace_common_fnc_setDead;
+ *
+ * Public: Yes
+ */
+
+params [["_unit", objNull, [objNull]], ["_reason", "", [""]], ["_source", objNull, [objNull]], ["_instigator", objNull, [objNull]]];
+
+if (!local _unit) exitWith {};
```
Comment (johnb432) [inline addons/common/functions/fnc_setDead.sqf:23]: Do we want to add a warning?

**Comment by johnb432** [comment]
Do we want to add a callable event for this, so that units can be killed from any client?

File: addons/common/functions/fnc_setDead.sqf (hunk 1 above)
Comment (rautamiekka) [reply addons/common/functions/fnc_setDead.sqf:23]: I don't see why not.

**Comment by PabstMirror** [comment]
[in ace_medical_statemachine_fnc_enteredStateDeath ](https://github.com/acemod/ACE3/blob/e535988479976f471663cc5ef137c58e220cfc33/addons/medical_statemachine/functions/fnc_enteredStateDeath.sqf#L28-L30)

we could add
`private _source = _unit getVariable [QEGVAR(medical,lastDamageSource), objNull];` 
and change to 
`[_unit, _causeOfDeath, _source, _instigator] call EFUNC(medical_status,setDead);`

should give us a more accurate killed EH, but I'm not sure what side effects it might have
or why we didn't do it originally

**Review by PabstMirror — APPROVED** [review]

**Comment by PabstMirror** [comment]
maybe merge now and I can do that in a followup

**Review by BrettMayson — COMMENTED** [review]
Add an event at the same time?

**Comment by BrettMayson** [comment]
That only fires with medical enabled right? And it also won't include the reason string

**Comment by johnb432** [comment]
> That only fires with medical enabled right? And it also won't include the reason string

`ace_killed` is indeed part of `medical_status`, but it includes the reason string (it's the 2nd param of `ace_killed`). It's raised automatically as the unit is killed, so there is no need to raise any events for when `ace_medical` is loaded.

I'm not sure if we want to add events when `ace_medical` isn't loaded. We *could* reuse `ace_killed`, but I'm not sure if that's a good idea. If a mod doesn't use `ace_medical`, it has to rely on the Vanilla `Killed` EH, meaning it could also have its own way of handling reasons of death.

I added the reason as a param for when `ace_medical` is loaded, I didn't intend it to be used when it wasn't loaded, as I don't see the point of using a reason when not using `ace_medical`.

**Comment by johnb432** [comment]
We can add events in another PR if it's necessary.

**Pull request merged**